invenio_vocabularies/datastreams/writers.py

```diff
@@ -87,23 +87,8 @@ def _resolve(self, id_):
     def _do_update(self, entry):
         vocab_id = self._entry_id(entry)
         current = self._resolve(vocab_id)
-        combined_dict = current.to_dict()
-
-        # Update fields from entry
-        for key, value in entry.items():
-            if key in combined_dict:
-                if isinstance(combined_dict[key], list) and isinstance(value, list):
-                    combined_dict[key].extend(
-                        item for item in value if item not in combined_dict[key]
-                    )
-                else:
-                    combined_dict[key] = value
-            else:
-                combined_dict[key] = value
-
-        return StreamEntry(
-            self._service.update(self._identity, vocab_id, combined_dict)
-        )
+        updated = dict(current.to_dict(), **entry)
+        return StreamEntry(self._service.update(self._identity, vocab_id, updated))
```

Comment:
Rolling back to the initial implementation (which overwrites keys at the top level), which was working for all cases, except for the OpenAIRE Affiliations, were we want to update/add identifiers to a list.